src/manager/layoutManager.ts

```diff
@@ -20,21 +20,35 @@ import { SimpleHorizontalLayout } from 'src/layout/msWorkspace/tilingLayouts/cus
 import { SimpleVerticalLayout } from 'src/layout/msWorkspace/tilingLayouts/custom/simpleVertical';
 import { RatioLayout } from 'src/layout/msWorkspace/tilingLayouts/custom/ratio';
 import { GridLayout } from 'src/layout/msWorkspace/tilingLayouts/custom/grid';
+import { MsWorkspace } from 'src/layout/msWorkspace/msWorkspace';
+import { Settings } from 'gio';
+
+type ExtractState<L> = L extends BaseTilingLayout<infer S> ? S : never;
```

Comment:
This is foreign language for me ^^

Reply:
You can think of it as a function. It maps `ExtractState<BaseTilingLayout<S>>` to `S`. It allows us to get the generic state type from a layout type.